testing/tests/DevExpress.ui.widgets/listParts/commonTests.js

```diff
@@ -1340,46 +1340,65 @@ QUnit.module('options changed', moduleSetup, () => {
         assert.strictEqual($list.find(`.${LIST_SELECT_ALL_LABEL_CLASS}`).text(), 'Select All');
     });
 
-    QUnit.test('menuItems', function(assert) {
-        const actionSpy = sinon.spy();
-        const menuItems = [{ text: 'action', action: actionSpy }];
-
+    QUnit.test('menuItems can be added runtime', function(assert) {
+        const menuItems = [{ text: 'action' }];
         const $list = $('#list').dxList({
             items: [{ text: 'test 1' }, { text: 'test 2' }],
             displayExpr: 'text',
             menuMode: 'context',
-            menuItems
+            menuItems: []
         });
-
-        let $item = $list.find('.dx-list-item').eq(0);
         const list = $list.dxList('instance');
+        list.option('menuItems', menuItems);
 
-        let contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
+        const $item = $list.find('.dx-list-item').eq(0);
+        const contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
         $item.trigger(contextMenuEvent);
 
-        let $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+        const $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
         assert.strictEqual($menuItems.length, 1, 'items count is correct');
-        $menuItems.eq(0).trigger('dxclick');
-        assert.strictEqual(actionSpy.callCount, 1, 'action is called once');
+    });
 
+    QUnit.test('menuItems can be changed to empty array', function(assert) {
+        const menuItems = [{ text: 'action' }];
+        const $list = $('#list').dxList({
+            items: [{ text: 'test 1' }, { text: 'test 2' }],
+            displayExpr: 'text',
+            menuMode: 'context',
+            menuItems
+        });
+        const list = $list.dxList('instance');
         list.option('menuItems', []);
-        $item = $list.find('.dx-list-item').eq(0);
-        contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
+
+        const $item = $list.find('.dx-list-item').eq(0);
+        const contextMenuEvent = $.Event('contextmenu', { pointerType: 'mouse' });
         $item.trigger(contextMenuEvent);
-        $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
+
+        const $menuItems = $(`.${LIST_CONTEXT_MENUCONTENT_CLASS}`).find('.dx-list-item');
         assert.strictEqual($menuItems.length, 0, 'items count is correct');
+    });
 
+    QUnit.test('menuItems can be changed to empty array and then back', function(assert) {
```

Comment:
It looks like this test duplicates the previous two ones. Do we need it?

Reply:
I think yes, we do. It is rare, but i remember some issues in similar cases

Reply:
> i remember some issues in similar cases

Any examples?